libraries/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -704,37 +696,31 @@ static IotMqttError_t _subscriptionCreateAndSerialize( IotMqttOperationType_t op
                                        pCallbackInfo,
                                        ppSubscriptionOperation );
 
-    if( status != IOT_MQTT_SUCCESS )
-    {
-        goto cleanup;
-    }
-    else
+    if( status == IOT_MQTT_SUCCESS )
     {
         pSubscriptionOperation = ( *ppSubscriptionOperation );
-    }
 
-    /* Check the subscription operation data and set the operation type. */
-    IotMqtt_Assert( pSubscriptionOperation->u.operation.status == IOT_MQTT_STATUS_PENDING );
-    IotMqtt_Assert( pSubscriptionOperation->u.operation.periodic.retry.limit == 0 );
-    pSubscriptionOperation->u.operation.type = operation;
 
-    /* Generate a subscription packet from the subscription list. */
-    status = serializeSubscription( pSubscriptionList,
-                                    subscriptionCount,
-                                    &( pSubscriptionOperation->u.operation.pMqttPacket ),
-                                    &( pSubscriptionOperation->u.operation.packetSize ),
-                                    &( pSubscriptionOperation->u.operation.packetIdentifier ) );
+        /* Check the subscription operation data and set the operation type. */
```

Comment:
This change is reordering this logic in addition to removing the goto. What's the rationale for the former?

Reply:
Agreed, thank you.